.idea/kotlinc.xml

```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
+  <component name="KotlinJpsPluginSettings">
+    <option name="version" value="1.6.21" />
```

Comment:
Where does this come from? 👀 

(When opening Terasology in IntelliJ on my Mac I got a warning about some old Kotlin version 1.4.x, and that it switched to use 1.6.21. I did not find any references to the specific version it warned about, though. Does this config file pin the version to something reasonable?)

@keturn any idea? IntelliJ help on [How to manage projects under Version Control Systems](https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems) seems to suggest that the file can be shared...

> What needs to be shared:
>
> 1. All files under the .idea directory in the project root except the items that store user-specific settings: ...

Reply:
lets separate this out; having it in here is making it more complicated to switch PR branches. #5068